<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>711</width>
    <height>496</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QTabWidget" name="MainTab">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="MainView">
       <attribute name="title">
        <string>Main</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="MainLayout">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QToolBox" name="ToolBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>159</width>
              <height>0</height>
             </size>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="PortConfiguration">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>186</width>
               <height>277</height>
              </rect>
             </property>
             <attribute name="label">
              <string>Port</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="4" column="0">
               <widget class="QLabel" name="DeviceInfoLabel">
                <property name="text">
                 <string>Device Info:</string>
                </property>
               </widget>
              </item>
              <item row="5" column="0">
               <widget class="QLabel" name="DeviceDescription">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <layout class="QVBoxLayout" name="DeviceLayout">
                <item>
                 <widget class="QLabel" name="DeviceLabel">
                  <property name="text">
                   <string>Device</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QComboBox" name="DeviceComboBox"/>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="RefreshDevices">
                <property name="text">
                 <string>Refresh</string>
                </property>
               </widget>
              </item>
              <item row="7" column="0">
               <widget class="QLabel" name="DevicePermissionsLabel">
                <property name="text">
                 <string>User Permissions:</string>
                </property>
               </widget>
              </item>
              <item row="6" column="0">
               <spacer name="verticalSpacer_8">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Fixed</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>0</width>
                  <height>15</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="8" column="0">
               <layout class="QHBoxLayout" name="DevicePermissionsLayout">
                <item>
                 <widget class="QLabel" name="DeviceReadPermission">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="DeviceWritePermission">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="3" column="0">
               <spacer name="verticalSpacer_9">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Fixed</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>0</width>
                  <height>15</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="11" column="0">
               <layout class="QHBoxLayout" name="DeviceOpenCloseLayout">
                <item>
                 <widget class="QPushButton" name="DeviceOpen">
                  <property name="text">
                   <string>Open</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="DeviceClose">
                  <property name="text">
                   <string>Close</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="12" column="0">
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>45</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="9" column="0">
               <spacer name="verticalSpacer_10">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Fixed</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>0</width>
                  <height>15</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="ChannelConfiguration">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>159</width>
               <height>272</height>
              </rect>
             </property>
             <attribute name="label">
              <string>Channel</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <layout class="QVBoxLayout" name="BaudRateLayout">
                <item>
                 <widget class="QLabel" name="BaudRateLabel">
                  <property name="text">
                   <string>Baud Rate</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QComboBox" name="BaudRateComboBox"/>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>173</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="CommunicationConfiguration">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>159</width>
               <height>272</height>
              </rect>
             </property>
             <attribute name="label">
              <string>Communication</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_5">
              <item row="0" column="0">
               <layout class="QVBoxLayout" name="CommunicationLayout">
                <item>
                 <widget class="QCheckBox" name="LocalEchoCheckBox">
                  <property name="text">
                   <string>Local Echo</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="CR_LF_CheckBox">
                  <property name="text">
                   <string>CR LF auto</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_7">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeType">
                   <enum>QSizePolicy::Fixed</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>0</width>
                    <height>10</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QLabel" name="ControlSignalsLayout">
                  <property name="text">
                   <string>Control Signals</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="SendBreakButton">
                  <property name="text">
                   <string>Send Break</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="ToggleDTRButton">
                  <property name="text">
                   <string>Toggle DTR</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="ToggleRTSButton">
                  <property name="text">
                   <string>Toggle RTS</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <spacer name="verticalSpacer_6">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>45</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="Status">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>159</width>
               <height>272</height>
              </rect>
             </property>
             <attribute name="label">
              <string>Status</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="0" column="0">
               <layout class="QVBoxLayout" name="StatusLayout">
                <item>
                 <widget class="QLabel" name="PortLabel">
                  <property name="text">
                   <string>Port</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="PortStatusLabel">
                  <property name="text">
                   <string>OK</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_4">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeType">
                   <enum>QSizePolicy::Fixed</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>92</width>
                    <height>13</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QLabel" name="CommunicationLabel">
                  <property name="text">
                   <string>Communication</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="CommunicationStatusLabel">
                  <property name="text">
                   <string>OK</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_5">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeType">
                   <enum>QSizePolicy::Fixed</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>92</width>
                    <height>13</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QLabel" name="MessageLabel">
                  <property name="text">
                   <string>Message</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="MessageStatusLabel">
                  <property name="text">
                   <string>OK</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_15">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeType">
                   <enum>QSizePolicy::Fixed</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>92</width>
                    <height>13</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="DTR_RTS_CTS_Layout">
                  <item>
                   <widget class="QLabel" name="DTRLabel">
                    <property name="text">
                     <string>DTR</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="RTSLabel">
                    <property name="text">
                     <string>RTS</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="CTSLabel">
                    <property name="text">
                     <string>CTS</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="CD_DSR_RI_Layout">
                  <item>
                   <widget class="QLabel" name="CDLabel">
                    <property name="text">
                     <string>CD</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="DSRLabel">
                    <property name="text">
                     <string>DSR</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="RILabel">
                    <property name="text">
                     <string>RI</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>5</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="InOut">
            <property name="sizeConstraint">
             <enum>QLayout::SetDefaultConstraint</enum>
            </property>
            <item>
             <layout class="QHBoxLayout" name="Output">
              <property name="sizeConstraint">
               <enum>QLayout::SetDefaultConstraint</enum>
              </property>
              <item>
               <widget class="QPlainTextEdit" name="OutputASCII">
                <property name="font">
                 <font>
                  <family>Monospace</family>
                 </font>
                </property>
                <property name="autoFillBackground">
                 <bool>false</bool>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
                <property name="textInteractionFlags">
                 <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPlainTextEdit" name="OutputHex">
                <property name="font">
                 <font>
                  <family>Monospace</family>
                 </font>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
                <property name="textInteractionFlags">
                 <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="Input">
              <property name="sizeConstraint">
               <enum>QLayout::SetDefaultConstraint</enum>
              </property>
              <item>
               <widget class="QCheckBox" name="HexInput">
                <property name="text">
                 <string>Hex</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">buttonGroup</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="DecInput">
                <property name="text">
                 <string>Dec</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">buttonGroup</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="OctInput">
                <property name="text">
                 <string>Oct</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">buttonGroup</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="BinInput">
                <property name="text">
                 <string>Bin</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">buttonGroup</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="ASCIIInput">
                <property name="text">
                 <string>ASCII</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">buttonGroup</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="InputText"/>
              </item>
              <item>
               <widget class="QPushButton" name="SendButton">
                <property name="text">
                 <string>Send</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="GraphicalView">
       <attribute name="title">
        <string>Graphical</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>711</width>
     <height>20</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionClear_screen"/>
    <addaction name="separator"/>
    <addaction name="actionSend_raw_file"/>
    <addaction name="actionSave_raw_file"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>?</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCopy"/>
    <addaction name="actionCopy_All"/>
    <addaction name="actionPaste"/>
   </widget>
   <widget class="QMenu" name="menuLog">
    <property name="title">
     <string>Log</string>
    </property>
    <addaction name="actionTo_File"/>
    <addaction name="separator"/>
    <addaction name="actionPause"/>
    <addaction name="actionStop"/>
    <addaction name="separator"/>
    <addaction name="actionClear"/>
   </widget>
   <widget class="QMenu" name="menuConfiguration">
    <property name="title">
     <string>Configuration</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuLog"/>
   <addaction name="menuConfiguration"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionSend_raw_file">
   <property name="text">
    <string>Send File</string>
   </property>
  </action>
  <action name="actionSave_raw_file">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionClear_screen">
   <property name="text">
    <string>Clear Screen</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>Paste</string>
   </property>
  </action>
  <action name="actionCopy_All">
   <property name="text">
    <string>Copy All</string>
   </property>
  </action>
  <action name="actionTo_File">
   <property name="text">
    <string>To File</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="text">
    <string>Pause</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clean</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
